packages/vite/src/node/plugins/ensureWatch.ts

```diff
@@ -0,0 +1,17 @@
+import type { Plugin } from '../plugin'
+import { cleanUrl, queryRE } from '../utils'
+
+/**
+ * plugin to ensure rollup can watch correctly.
+ */
+export function ensureWatchPlugin(): Plugin {
+  return {
+    name: 'vite:ensure-watch',
+    load(id) {
+      if (queryRE.test(id)) {
+        this.addWatchFile(cleanUrl(id))
+      }
+      return null
+    }
```

Comment:
Just have a thought about this. If the id is not an actual file path, but matches the `queryRE`, would this be safe to call?

I think a couple cases where this could happen is e.g. path aliases and virtual modules.

Reply:
Thanks for review!

> I think a couple cases where this could happen is e.g. path aliases and virtual modules.

In the current implementation,the virtual modules and path alias already exists in the rollup watchFile list.What i do is adding the correct id in it.

Reply:
And in rollup, `chokidar` has own implementation to filter the incorrect id.IMO, it would be safe to call it.